vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -78,23 +98,46 @@ def forward(
         # Only perform shape/dtype/device checking in strict mode, as it adds
         # overhead.
         if self._strict_mode:
-            self._raise_if_incorrect_input(target_probs, draft_token_ids,
-                                           bonus_token_ids, draft_probs)
+            self._raise_if_incorrect_input(target_with_bonus_probs,
+                                           draft_token_ids, bonus_token_ids,
+                                           draft_probs)
 
-        accepted, recovered_token_ids = (
-            self._batch_modified_rejection_sampling(
-                target_probs,
-                draft_probs,
-                draft_token_ids,
-                seeded_seqs,
-            ))
+        batch_size, k, _ = draft_probs.shape
 
-        output_token_ids = self._create_output(
-            accepted,
-            recovered_token_ids,
-            draft_token_ids,
-            bonus_token_ids,
-        )
+        if batch_size == 0:
+            return torch.empty(0, k + 1, device=draft_probs.device, dtype=int)
+
+        # If use Flashinfer chain_speculative_sampling kernel
+        # for rejection sampling
+        if chain_speculative_sampling is not None and self.use_flashinfer:
```

Comment:
code readability: it is good practice to reduce the number of conditions available in the code, both so it's easier to read and so there are less bugs due to less possible permutations of state.

in this case, suggest having a single condition `self.use_flashinfer`, which is validated in the constructor (e.g. if use_flashinfer is True, then assert chain_speculative_sampling is not None). this will simplify the code and reduce the amount of state.